tokio/src/runtime/thread_id.rs

```diff
@@ -0,0 +1,31 @@
+use std::num::NonZeroU64;
+
+#[derive(Eq, PartialEq, Clone, Copy, Hash, Debug)]
+pub(crate) struct ThreadId(NonZeroU64);
+
+impl ThreadId {
+    pub(crate) fn next() -> Self {
+        use crate::loom::sync::atomic::{Ordering::Relaxed, StaticAtomicU64};
+
+        static NEXT_ID: StaticAtomicU64 = StaticAtomicU64::new(0);
+
+        let mut last = NEXT_ID.load(Relaxed);
+        loop {
+            let id = match last.checked_add(1) {
+                Some(id) => id,
+                None => exhausted(),
+            };
+
+            match NEXT_ID.compare_exchange_weak(last, id, Relaxed, Relaxed) {
+                Ok(_) => return ThreadId(NonZeroU64::new(id).unwrap()),
```

Comment:
The unwrap is a bit unfortunate. One option is to define a method call like this one:
```Rust
fn checked_add_one(num: u64) -> Option<NonZeroU64> {
    NonZeroU64::new(num.wrapping_add(1))
}
```
That said, it is compiled out in this instance, so I am also okay with keeping it as-is.

Reply:
I grabbed it from `std` here :shrug: